### PR TITLE
[Windows] Make sure the InputMethod deletes current selection before preedit is set

### PR DESCRIPTION
**Pull request by Gillibald**
## What does the pull request do?

If there is a selection the InputMethod should delete it when a new composition starts otherwise the selection isn't deleted until the composition is committed.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0036694-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

File: src/Windows/Avalonia.Win32/Input/Imm32InputMethod.cs [hunk 1]
```diff
@@ -324,6 +324,11 @@ public void HandleCompositionStart()
             if (IsActive)
             {
                 Client.SetPreeditText(null);
+
+                if (Client.SupportsSurroundingText && Client.SurroundingText.AnchorOffset != Client.SurroundingText.CursorOffset)
+                {
+                    KeyPress(Key.Delete);
```
Comment (kekekeks) [inline src/Windows/Avalonia.Win32/Input/Imm32InputMethod.cs:330]: Simulating actions via sending events feels a bit like a hack. Could we issue a text input event with empty text instead? Or perhaps introduce a separate method while we still can?

Comment (Gillibald) [reply src/Windows/Avalonia.Win32/Input/Imm32InputMethod.cs:330]: The Mac backend makes the exact same thing. The input context sends a delete key and we process it. But y we could just send an TextInputEvent with an empty string.

Comment (Gillibald) [reply src/Windows/Avalonia.Win32/Input/Imm32InputMethod.cs:330]: Same for iOS. Ideally deletion could be processed by the TextInputMethodClient so we have a shared implementation for all platforms.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0037145-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Review by kekekeks — APPROVED** [review]

**Pull request merged**